src/transformStateWithClones.js

```diff
@@ -6,8 +6,41 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const rez = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      rez.push({});
+    }
+
+    if (actions[i].type === 'addProperties' && rez.length > 0) {
```

Comment:
```suggestion
    if (actions[i].type === 'addProperties') {
```

Reply:
Consider using the switch here

Reply:
Also, you can use a helper object to change it instead of checking the size of the array